be/src/exec/orc_scanner.cpp

```diff
@@ -0,0 +1,298 @@
+                            LOG(WARNING) << str_error.str();
+                            return Status::InternalError(str_error.str());
+                        }
+                    }
+                }
+            }
+            ++_current_line_of_group;
+
+            // range of current file
+            const TBrokerRangeDesc &range = _ranges.at(_next_range - 1);
+            if (range.__isset.num_of_columns_from_file) {
+                fill_slots_of_columns_from_path(range.num_of_columns_from_file, range.columns_from_path);
+            }
+            COUNTER_UPDATE(_rows_read_counter, 1);
+            SCOPED_TIMER(_materialize_timer);
+            if (fill_dest_tuple(Slice(), tuple, tuple_pool)) {
+                break; // get one line, break from while
+            } // else skip this line and continue get_next to return
+        }
+        return Status::OK();
+    } catch (orc::ParseError &e) {
+        std::stringstream str_error;
+        str_error << "ParseError : " << e.what();
+        LOG(WARNING) << str_error.str();
+        return Status::InternalError(str_error.str());
+    } catch (orc::InvalidArgument &e) {
+        std::stringstream str_error;
+        str_error << "ParseError : " << e.what();
+        LOG(WARNING) << str_error.str();
+        return Status::InternalError(str_error.str());
+    }
+}
+
+Status ORCScanner::open_next_reader() {
+    while (true) {
+        if (_next_range >= _ranges.size()) {
+            _scanner_eof = true;
+            return Status::OK();
+        }
+        const TBrokerRangeDesc &range = _ranges[_next_range++];
+        std::unique_ptr<FileReader> file_reader;
+        switch (range.file_type) {
+            case TFileType::FILE_LOCAL: {
+                file_reader.reset(new LocalFileReader(range.path, range.start_offset));
+                break;
+            }
+            case TFileType::FILE_BROKER: {
+                int64_t file_size = 0;
+                // for compatibility
+                if (range.__isset.file_size) { file_size = range.file_size; }
+                file_reader.reset(new BrokerReader(_state->exec_env(), _broker_addresses, _params.properties,
+                                                   range.path, range.start_offset, file_size));
+                break;
+            }
+            default: {
+                std::stringstream ss;
+                ss << "Unknown file type, type=" << range.file_type;
+                return Status::InternalError(ss.str());
+            }
+        }
+        RETURN_IF_ERROR(file_reader->open());
+        if (file_reader->size() == 0) {
+            file_reader->close();
+            continue;
+        }
+
+        std::unique_ptr<orc::InputStream> inStream = std::unique_ptr<orc::InputStream>(
+                new ORCFileStream(file_reader.release(), range.path));
+        _reader = orc::createReader(std::move(inStream), _options);
```

Comment:
propose orc file reader add new function to realize，3ks

Reply:
Hi, why add new function?